### PR TITLE
Support to SD-JWT credentials

### PR DESCRIPTION
**Pull request by Juliano1612**
Depends on https://github.com/spruceid/mobile-sdk-rs/pull/46 and https://github.com/spruceid/mobile-sdk-swift/pull/34

This enables the reference app to work with SD-JWTs and creates a new preview + "add to wallet" flow.

### CODE REVIEW

File: project.yml [hunk 1]
```diff
@@ -3,10 +3,10 @@ options:
   bundleIdPrefix: com.spruceid.mobile
 packages:
   SpruceIDMobileSdk:
-    url: https://github.com/spruceid/mobile-sdk-swift
+    # url: https://github.com/spruceid/mobile-sdk-swift
     # from: 0.0.10
-    revision: 8b80370
-    # path: "../mobile-sdk-swift"
+    # revision: 8b80370
+    path: "../mobile-sdk-swift"
```
Comment (Juliano1612) [inline project.yml:9]: Pointing to local dependency

File: project.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ packages:
   SpruceIDMobileSdk:
     url: https://github.com/spruceid/mobile-sdk-swift
     # from: 0.0.10
-    revision: 8b80370
+    revision: 379b4a3d4550df6ab9da0e5be3406d8021dcd253
```
Comment (Juliano1612) [inline project.yml:8]: Pointing to the last commit on https://github.com/spruceid/mobile-sdk-swift/pull/34

File: Targets/AppUIKit/Sources/dataStores/CredentialDataStore.swift
Comment (Ryanmtate) [inline Targets/AppUIKit/Sources/dataStores/CredentialDataStore.swift:1]: This is more of a question/comment than a requested change.

Can this datastore eventually serve as the storage engine for the `StorageInterface` that we export from mobile-sdk-rs?

Are there any considerations for where/how the credentials should be stored, e.g. encrypted, etc. That we can explore in the future?

This looks fine to me for providing immediate support for persisting credentials, as long as we have a path to using the VDC collection when stable.

Maybe worth opening some linear tickets for these items, if we aren't already tracking.

**Review by Ryanmtate — APPROVED** [review]
@Juliano1612 Nice work! I've got the reference app running and it looks great!

I haven't tested the QR code scanning, but would love to understand how to get an e2e demo working.

File: Targets/AppUIKit/Sources/dataStores/CredentialDataStore.swift
Comment (Juliano1612) [reply Targets/AppUIKit/Sources/dataStores/CredentialDataStore.swift:1]: This datastore is the configuration to store data using SQLite. We already have it configured for generic verification logs, so I added the credentials to simplify it. As far as I know, we haven't discussed the considerations about storing credentials yet. 

You can see it as an SQL database on the front end, so we can run SQL queries or use it through the ORM. I believe that using this (for both Swift and Kotlin), we don't need to reinvent the wheel to manage stored data.

**Pull request merged**